proposals/csharp-8.0/ranges.md

```diff
@@ -0,0 +1,58 @@
+# Ranges
+
+* [x] Proposed
+* [ ] Prototype: Not Started
+* [ ] Implementation: Not Started
+* [ ] Specification: Not Started
+
+## Summary
+
+This feature is about delivering two new operators that allow constructing `System.Index` and `System.Range` objects, and using them to index/slice collections at runtime.
+
+## Detailed Design
+
+#### System.Index
+
+C# has no way of indexing a collection from the end, but rather most indexers use the "from start" notion, or do a "length - i" expression. We introduce a new Index expression that means "from the end". The feature will introduce a new unary prefix "hat" operator. Its single operand must be convertible to `System.Int32`. It will be lowered into the appropriate `System.Index` factory method call.
```

Comment:
It's probably worth noting that the `Index` type will have an implicit conversion from `Int32`, which would throw for negative values.
If that's not correct, let's take an open LDM issue.